### PR TITLE
User exp

### LINKED ISSUE #92: Implement xp fetching for use by UserController

### PR DESCRIPTION
**Pull request by tuesmiddt**
Fixes #92

### CODE REVIEW

File: lib/cadet/assessments/assessments.ex [hunk 1]
```diff
@@ -13,6 +13,22 @@ defmodule Cadet.Assessments do
   @submit_answer_roles ~w(student)a
   @grading_roles ~w(staff)a
 
+  def user_total_xp(user = %User{role: role}) do
```
Comment (indocomsoft) [inline lib/cadet/assessments/assessments.ex]: Why do you bind variable `role` haha

File: lib/cadet/assessments/assessments.ex [hunk 2]
```diff
@@ -13,6 +13,22 @@ defmodule Cadet.Assessments do
...
+    xp =
+      Query.all_submissions_with_xp()
+      |> subquery()
+      |> group_by(:student_id)
+      |> where(student_id: ^user.id)
+      |> select([s], sum(s.xp))
+      |> Repo.one()
```
Comment (indocomsoft) [inline lib/cadet/assessments/assessments.ex]: Prefer
```elixir
Query.all_submissions_with_xp()
|> subquery()
|> where(student_id: ^user.id)
|> Repo.aggregate(:sum, :xp)
```

File: test/cadet_web/controllers/user_controller_test.exs [hunk 3]
```diff
@@ -12,8 +12,23 @@ defmodule CadetWeb.UserControllerTest do
   end
 
   describe "GET /user" do
-    test "success", %{conn: conn} do
+    test "success, student", %{conn: conn} do
```
Comment (indocomsoft) [inline test/cadet_web/controllers/user_controller_test.exs]: My bad, I think I missed this in my code review of the PR. Could you help me refactor this file to use tags instead?

```elixir
@tags authenticate: :student
```
Similarly for the staff below.

Comment (tuesmiddt) [reply test/cadet_web/controllers/user_controller_test.exs]: Done

**Pull request merged**